src/transform-sdk/rust/core-types/src/lib.rs

```diff
@@ -116,13 +116,36 @@ impl<'a> From<&'a WrittenRecord<'a>> for BorrowedRecord<'a> {
     }
 }
 
+/// WriteOptions allows for customizing a RecordWriter's write.
```

Comment:
I'm guessing that WriteOptions and RecordWriter are both functions that the user can use?

Reply:
Yes good idea, let me do this.

Reply:
Much better

<img width="684" alt="Screenshot 2024-03-19 at 1 46 09 PM" src="https://github.com/redpanda-data/redpanda/assets/8669100/a0a64d0b-b84f-43f9-b2d6-3561b0b6d224">